addons/project/models/project.py

```diff
@@ -1345,6 +1350,14 @@ def _search_is_private(self, operator, value):
         else:
             return [('project_id', '!=', False)]
 
+    @api.model
+    def _get_view(self, view_id=None, view_type='form', **options):
+        arch, view = super()._get_view(view_id, view_type, **options)
+        if view_type == 'search' and  self.env.user.notification_type == 'email':
+            for node in arch.xpath("//filter[@name='message_needaction']"):
+                node.set('invisible', '1')
```

Comment:
@tde-banana-odoo couldn't this be generic in mail? The idea is to set the `message_needaction` filter invisible if the user has not set its notification is being handled in odoo. Thanks for your insight 🙂